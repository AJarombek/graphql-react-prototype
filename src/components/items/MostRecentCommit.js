/**
 * MostRecentCommit component which displays repositories that have been contributed to most
 * recently.
 * @author Andrew Jarombek
 * @since 4/21/2020
 */

import React, { useEffect, useState } from 'react';
import moment from 'moment';
import { getMostRecentCommit } from '../../datasource/GraphQL';

const MostRecentCommit = () => {
  const [reposByDate, setReposByDate] = useState([]);
  const [error, setError] = useState(null);

  const generateMostRecentCommits = (repositories) => {
    const repositoriesByDate = [];

    // eslint-disable-next-line no-restricted-syntax
    for (const repository of repositories) {
      repositoriesByDate.push({
        name: repository.node.name,
        lastCommitDate: moment(repository.node?.ref?.target?.pushedDate),
      });
    }

    setReposByDate(repositoriesByDate);
    setError(null);
  };

  useEffect(() => {
    async function getGraphQLResult() {
      const result = await getMostRecentCommit('AJarombek');

      if (result.data.data) {
        generateMostRecentCommits(result.data.data.user.repositories.edges);
        setError(null);
      } else {
        setError(result.data.errors[0].message);
      }
    }

    getGraphQLResult();
  }, []);

  return (
    <div className="items most-recent-commit">
      {error ? (
        <div className="error">
          <h6>{error}</h6>
        </div>
      ) : (
        <>
          <h2>Most Recently Committed</h2>
          {reposByDate.map((repository) => (
            <div className="commit" key={repository.name}>
              <p>{repository.name}</p>
              <p>{repository.lastCommitDate.format('MMM Do, YYYY h:mm A')}</p>
            </div>
          ))}
        </>
      )}
    </div>
  );
};

export default MostRecentCommit;
