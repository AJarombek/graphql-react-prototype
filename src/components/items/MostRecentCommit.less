/**
 * Stylesheet for the MostRecentCommit component.
 * Author: Andrew Jarombek
 * Date: 4/21/2020
 */

@import "../../vars";

.most-recent-commit {
  grid-area: recCommit;

  h2 {
    margin-bottom: 20px;
  }

  .commit:nth-child(2) {
    p:nth-child(2) {
      .hoverPurple();
    }
  }

  .commit {
    display: flex;
    align-items: center;
    margin: 8px 0;

    p:nth-child(1) {
      font-size: 16px;

      @media (max-width: 1231px) {
        font-size: 14px;
      }
    }

    p:nth-child(2) {
      font-size: 18px;
      margin: 0 0 0 auto;

      @media (max-width: 1231px) {
        font-size: 16px;
      }
    }
  }
}
