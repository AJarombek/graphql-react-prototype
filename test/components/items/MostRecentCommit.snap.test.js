/**
 * Snapshot test for the MostRecentCommit component.
 * @author Andrew Jarombek
 * @since 4/25/2020
 */

import React from 'react';
import MostRecentCommit from '../../../src/components/items/MostRecentCommit';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<MostRecentCommit />).toJSON();
  expect(tree).toMatchSnapshot();
});
