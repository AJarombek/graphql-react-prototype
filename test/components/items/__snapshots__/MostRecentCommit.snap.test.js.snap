// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="items most-recent-commit"
>
  <h2>
    Most Total Commits
  </h2>
</div>
`;
